src/azure-cli-core/azure/cli/core/commands/command_operation.py

```diff
@@ -0,0 +1,509 @@
+        import types
+
+        from azure.cli.core.profiles import AZURE_API_PROFILES
+        from azure.cli.core.profiles._shared import get_versioned_sdk_path
+
+        for rt in AZURE_API_PROFILES[self.cli_ctx.cloud.profile]:
+            if op_path.startswith(rt.import_prefix + '.'):
+                op_path = op_path.replace(rt.import_prefix,
+                                          get_versioned_sdk_path(self.cli_ctx.cloud.profile, rt,
+                                                                 operation_group=self.operation_group))
+
+        try:
+            mod_to_import, attr_path = op_path.split('#')
+            handler = import_module(mod_to_import)
+            for part in attr_path.split('.'):
+                handler = getattr(handler, part)
+            if isinstance(handler, types.FunctionType):
+                return handler
+            return get_method_function(handler)
+        except (ValueError, AttributeError):
+            raise ValueError("The operation '{}' is invalid.".format(op_path))
+
+    def load_getter_op_arguments(self, getter_op_path, cmd_args=None):
+        """ Load arguments from function signature of getter command op """
+        op = self.get_op_handler(getter_op_path)
+        getter_args = dict(
+            extract_args_from_signature(op, excluded_params=EXCLUDED_PARAMS))
+        cmd_args = cmd_args or {}
+        cmd_args.update(getter_args)
+        # The cmd argument is required when calling self.handler function.
+        cmd_args['cmd'] = CLICommandArgument('cmd', arg_type=ignore_type)
+        return cmd_args
+
+    def apply_doc_string(self, handler):
+        return self.ctx._apply_doc_string(handler, self.kwargs)  # pylint: disable=protected-access
+
+    def load_op_handler_description(self, handler=None):
+        """ Load the description from function signature of command op """
+        if handler is None:
+            def default_handler():
+                """"""  # Use empty __doc__ property here, which is required in extract_full_summary_from_signature
+            handler = default_handler
+        self.apply_doc_string(handler)
+        return extract_full_summary_from_signature(handler)
+
+    def resolve_client_arg_name(self, op_path):
+        from azure.cli.core.commands.client_factory import resolve_client_arg_name
+        return resolve_client_arg_name(op_path, self.kwargs)
+
+
+class CommandOperation(BaseCommandOperation):
+
+    def __init__(self, ctx, op_path, **kwargs):
+        if not isinstance(op_path, string_types):
+            raise TypeError("Operation must be a string. Got '{}'".format(op_path))
+        super(CommandOperation, self).__init__(ctx, **kwargs)
+        self.op_path = op_path
+
+    def handler(self, command_args):
+        """ Callback function of CLICommand handler """
+        from azure.cli.core.util import get_arg_list, augment_no_wait_handler_args
+
+        op = self.get_op_handler(self.op_path)
```

Comment:
May I know why not honor `handler` from developer any more?
https://github.com/Azure/azure-cli/blob/d80da3b29904456e9bf6eb02eebd68ede0369f83/src/azure-cli-core/azure/cli/core/__init__.py#L798

And it seems your op_path misses operation group info.

Reply:
That's because `handler` is always `None` for _command function:

https://github.com/Azure/azure-cli/blob/d80da3b29904456e9bf6eb02eebd68ede0369f83/src/azure-cli-core/azure/cli/core/commands/__init__.py#L1249-L1260